vividus-extension-visual-testing/src/main/java/org/vividus/visual/steps/AbstractVisualSteps.java

```diff
@@ -43,11 +43,15 @@ protected AbstractVisualSteps(IUiContext uiContext, IAttachmentPublisher attachm
         this.softAssert = softAssert;
     }
 
-    protected <T extends VisualCheck> VisualCheckResult execute(Function<T, VisualCheckResult> checkResultProvider,
+    protected <T extends VisualCheck> Optional<VisualCheckResult> execute(
+            Function<T, VisualCheckResult> checkResultProvider,
             Supplier<T> visualCheckFactory, String templateName)
     {
         SearchContext searchContext = uiContext.getSearchContext();
-        Validate.validState(searchContext != null, "Search context is null, please check is browser session started");
```

Comment:
```
return Optional.ofNullable(uiContext.getSearchContext(SearchContext.class)).map(sc -> {
    T visualCheck = visualCheckFactory.get();
        visualCheck.setSearchContext(searchContext);
       return visualCheck;
}).map(checkResultsProvider::apply).map(r -> {
           attachmentPublisher.publishAttachment(templateName, Map.of("result", result), "Visual comparison");
            verifyResult(result);
return result;
})
```

Reply:
For me original code looks more readable

Reply:
Yes, but there is already a method, which does validation in ui context, why not reuse it?